### DISCUSSION TITLE
[UI] The application takes a long time (minutes) to start.

### DISCUSSION THREAD

**Discussion by JoseOlin** [olive-editor/olive#1873] [General]
**Commit Hash** 

 76831a2b 

**Platform** 
Linux Mint Cinnamon 20.2

**Summary**
The problem is recent and I think it is related to having installed **QJackCtl**
Terminal show the last line of the output for several minutes, and eventually Olive starts

**Additional Information / Output**
Cannot connect to server socket err = File or directory does not exist
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
Cannot connect to server socket err = File or directory does not exist
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
[INFO] Setting output audio device to default ((null):0)
[INFO] Setting input audio device to default ((null):0)

**Comment by TheTechRobo** [comment]
It takes a few extra seconds for it to start up for me, but _minutes_? Weird.

**Comment by TheTechRobo** [comment]
Have you tried uninstalling QJackCtl, as you think that may be the issue?

**Comment by JoseOlin** [comment]
I just did and it took only a few seconds again. I'll reinstall it when I'm done editing to make sure it's the cause.

  The problem is that I use QJackCtl to check my audio levels before recording, so I'd like to make them co-exist. 

Thanks for that fast response! Amazing software btw!

**Comment by TheTechRobo** [comment]
I'm not the developer, I'm just a user of this great software :D

**Comment by JoseOlin** [comment]
Updating: QJackCtl is definitely not the problem.

Today it took **9 minutes**  to go from
[INFO] Setting input audio device to default ((null):0)

to
[DEBUG] Using graphics driver: Intel ((null):0)

The delay **only occurs the first time the AppImage is executed**; once Olive runs the first time, the **following ones open in a few seconds**. 

Nice to meet you, @TheTechRobo . A beautiful piece of software is being developed here.

**Comment by itsmattkc** [comment]
This appears to be specific to your setup in some way, I haven't experienced this on my end and other users don't appear to encounter it. From the output, it would appear to be either audio or OpenGL related. That `Using graphics driver` line is generated after creating an OpenGL context to check which driver is being used, so there might be something getting tripped up when first initializing OpenGL on your system.

**Comment by TheTechRobo** [comment]
since it only happens the first time, that could further the cause of this down more. it's probably a setting being set to the default.

**Comment by RafaelLinux** [comment]
I use just last AppImage (downloaded just now) and opens quickly without issues.

> I just did and it took only a few seconds again. I'll reinstall it when I'm done editing to make sure it's the cause.
> 
> The problem is that I use QJackCtl to check my audio levels before recording, so I'd like to make them co-exist.
> 
> Thanks for that fast response! Amazing software btw!

I have downloaded the latest AppImage from Olive and it launches without any problems.

For Jack related problems, your life would be much simpler if you forget about PulseAudio and install PipeWire and its libraries for Jack and Pulseaudio.

**Comment by JoseOlin** [comment]
>For Jack related problems, your life would be much simpler if you forget about PulseAudio and install PipeWire and its libraries for Jack and Pulseaudio.

I decided to try PipeWire (after trying almost everything proposed in [this thread](https://forums.linuxmint.com/viewtopic.php?t=302518)).

With Pipewire installed and running the messages in terminal change slightly  but the problem persists. The boot is delayed on the same line.

	[DEBUG] using qt5ct plugin ((null):0)
	[DEBUG] Extracting default OCIO config to "/home/user/.cache/olivevideoeditor.org/Olive/ocioconf" ((null):0)
	[INFO] Using Qt version: 5.12.8 ((null):0)
	ALSA lib pcm.c:2642:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
	ALSA lib pcm.c:2642:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
	ALSA lib pcm.c:2642:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
	ALSA lib pcm_route.c:869:(find_matching_chmap) Found no matching channel map
	ALSA lib pcm_route.c:869:(find_matching_chmap) Found no matching channel map
	ALSA lib pcm_route.c:869:(find_matching_chmap) Found no matching channel map
	ALSA lib pcm_route.c:869:(find_matching_chmap) Found no matching channel map
	ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
	ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
	ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
	ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
	[INFO] Setting output audio device to Audio Interno Estéreo analógico ((null):0)
	[INFO] Setting input audio device to Audio Interno Estéreo analógico ((null):0)

Even if doesn't solve this particular problem, PipeWire looks very promising as audio server. 
Thanks for the recommendation @RafaelLinux 

Something happens between, 

	[INFO] Setting input audio device to Audio Interno Estéreo analógico ((null):0)

and

	[DEBUG] D-Bus global menu: no ((null):0)

which survives during the session, but not after reboot.

Any config file that I can check before and after the first boot of Olive? 


> This appears to be specific to your setup in some way

Yes, it's very likely. 

<br>

> From the output, it would appear to be either audio or OpenGL related. That `Using graphics driver` line is generated after creating an OpenGL context to check which driver is being used.

Thanks, that helps me make a more specific search, and given that PulseAudio wasn't the problem I'll try with OpenGL. Any script I can run to narrow down the cause of the problem?

<br>

> I have downloaded the latest AppImage from Olive and it launches without any problems.

Yep, I'm sure it's not an AppImage specific issue since I've compiled the source code  (7d52dc19) and the same problem occurs. 


I've tried running Olive AppImage with gdb but it doesn't offer more information than the regular terminal. 

Any way to increase the verbosity? 


Thanks for all your support.

**Comment by RafaelLinux** [comment]
I guess you use Ubuntu or Debian, so I don't know if you have all needed applications and libraries to make work Pipewire. In openSUSE, I installed pipewire, wireplumber (before was pipewire-sessions), pipewire-pulseaudio, pipewire-libjack-0_3 and qpwgraph. You must uninstall pulseaudio and jack cause with mentioned libraries they are not neccesary. It's all I can tell you if your issue with Olive is related to audio conflicts.

**Comment by itsmattkc** [comment]
To rule out or confirm an OpenGL issue, try removing the following section (everything between `#ifdef Q_OS_LINUX` and `#endif`) and recompile: https://github.com/olive-editor/olive/blob/master/app/window/mainwindow/mainwindow.cpp#L833

I may convert this to a "discussion" since this seems to be more technical support and is likely specific to your setup. If an issue is confirmed in Olive, it can be converted back.

**Comment by JoseOlin** [comment]
Today I created a secondary user and ran Olive from there...it started without a problem from the secondary account, which made it clear that the problem wasn't the build, but some configuration in the user folders.

So I went back to my normal user and deleted
/home/user/.config/olivevideoeditor.org
/home/user/.cache/olivevideoeditor.org
/home/user/.local/share/olivevideoeditor.org             (This one was 11GB and was apparently the real culprit)

Then Olive started like a champ.

Problem Solved!
Thanks!
